arch/arm/core/cortex_m/fault.c

```diff
@@ -1113,7 +1113,7 @@ void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return,
 		"ESF could not be retrieved successfully. Shall never occur.");
 
 #ifdef CONFIG_DEBUG_COREDUMP
-	z_arm_coredump_fault_sp = POINTER_TO_UINT(esf);
```

Comment:
Since there's already a structure to hold the ESF, wouldn't it make sense to add SP to it directly rather than using a global variable instead ?

Reply:
It would have to go in `extra_info` since it's not part of the hardware ESF, and then it would *only* be available with `CONFIG_EXTRA_EXCEPTION_INFO=y`, even though it's pretty basic and critical information.

Reply:
is there anything preventing from having 
```c
#if defined(CONFIG_DEBUG_COREDUMP)
	uint32_t fault_sp;
#endif
```

in `struct arch_esf` between the FPU stack frame and the extra esf info?

Note I don't know where the coredump expect SP to be pointing at so I can only assume that your change is correct.
But this made me realise that this global variable unnecessarily reserves a word of memory for no use beyond the few milli(micro?) seconds the fault handler goes from `z_arm_fault` down to `coredump` when all other data are passed between them through `arch_esf`.

Reply:
You could probably do something like that, but it would require further changes in pretty critical code that I'm not familiar with, so I don't really dare to touch this for a single word of memory usage.